src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_ioctl.c

```diff
@@ -377,20 +378,36 @@ xocl_read_axlf_helper(struct xocl_drm *drm_p, struct drm_xocl_axlf *axlf_ptr)
 	void *ulp_blob;
 	void *kernels;
 	int rc;
+	struct xclErrorLast err_last;
 
+
+	err_last.pid = 0;
+	err_last.ts = 0; //TODO timestamp
 	if (!xocl_is_unified(xdev)) {
 		userpf_err(xdev, "XOCL: not unified Shell\n");
+		err_last.err_code = XRT_ERROR_CODE_BUILD(XRT_ERROR_NUM_XCLBIN_INCOMPATIBLE, 
+			XRT_ERROR_DRIVER_XOCL, XRT_ERROR_SEVERITY_CRITICAL, 
+			XRT_ERROR_MODULE_XCLBIN, XRT_ERROR_CLASS_SYSTEM);
+		xocl_insert_error_record(&xdev->core, &err_last);
```

Comment:
I think this is for a not supported shell, not for incompatible xclbin. @houlz0507 can point you to a more suitable location where xclbin compatibility is checked.

Reply:
all the xclbin checks on xocl side are in xocl_read_axlf_helper().

Reply:
updated